src/algorithms/tracking/TrackSeeding.cc

```diff
@@ -207,14 +207,19 @@ std::unique_ptr<edm4eic::TrackParametersCollection> eicrecon::TrackSeeding::make
       auto perigee = Acts::Surface::makeShared<Acts::PerigeeSurface>(Acts::Vector3(0,0,0));
       Acts::Vector3 global(xypos.first, xypos.second, z0);
 
-      auto local = perigee->globalToLocal(m_geoSvc->getActsGeometryContext(),
-                                          global, Acts::Vector3(1,1,1));
+      //Overwritten if call to globalToLocal is successful
+      Acts::Vector2 localpos(-9999., -9999.);
```

Comment:
No, this is potentially even worse. It is even more wrong than the sqrt(x,y) you had before.

The point is that you don't need this: if the conversion is bad, then don't generate a seed at all.

Reply:
It's not clear to me that if the conversion is bad we just shouldn't generate a seed from it. This could potentially remove highly displaced tracks from the seeding, i.e. those that are not compatible with a line surface at (0,0,0), which could be very secondary decays from Kshort decays or conversions or...

Reply:
By writing 9999, it is now a bad unphysical seed. Maybe you could do better for all the reasons Joe mentioned (and even the sqrt(x,y) was better), but writing 9999 is not right and in that case you are better off not writing any seed at all.

Reply:
I did a few checks. 

1) I generated 10k muon events with p = [0.5,20] GeV/c and eta = [-4,4] and phi = [-Pi,Pi] at each of the following settings and checked how often the `globalToLocal` succeeds. I did this with a 1 mm tolerance and the updated `direction` vector.

| Position (x,y,z) mm      | Number of seeds found | Number of seeds where `globalToLocal `fails
| ------------------------ | -------------------------- |------------------------------------------------
| (0,0,0)                          |       25774                       | 0 |
| (0,0,10)                        |       25482                       | 0 |
| (0,0,-10)                      |        25583                       | 0 |
| (1,0,0)                          |       25521                       | 0 |
| (10,0,0)                       |        5567                         | 0 |

So, for all these events, I never see a failure in `globalToLocal`.

2) For the case where the muon is generated from (0,0,0), I checked the Acts Loc-a distribution with a 0.1 um tolerance and a 1 mm tolerance. The results are identical:

0.1 micron tolerance:

<img width="629" alt="image_1" src="https://github.com/eic/EICrecon/assets/8701169/c4dab816-41f9-40cf-81bf-6d79719e78a5">

1 mm tolerance:

<img width="614" alt="image_2" src="https://github.com/eic/EICrecon/assets/8701169/a848453b-0a3c-4918-97ef-9b5fec14d839">

I would have expected to see an effect at 1 mm based on the range of the distribution.

3) As to the question of what default value to set before calling globalToLocal, maybe it doesn't matter based on the results above. But since Acts `Loc-a` will either be `+hypot(x,y)` or `-hypot(x,y)`, then we can either set the sign randomly to avoid any bias

`m_rng.uniform_double<double>(-1,1) < 0 ? -1 : 1`

Or we can set the sign based on the phi value. For example, for the case when I generate the particle from `(1,0,0) mm` and then plot the seed `phi` vs. seed Acts `Loc-a`, I see the following:

<img width="525" alt="image_3" src="https://github.com/eic/EICrecon/assets/8701169/67bcd442-15ec-4586-a1b9-2750716597de">

So, if `phi` is less than 0, the Acts Loc-a will be positive. I think this should hold in general, but I'm not completely sure.

Reply:
In this simulation, are all the tracks thrown from exactly (0,0,0) (or whatever vertex value you used), i.e. there is no vertex spread at all?

Reply:
Yes, that is correct. If I throw from `(x,y,z) = (0,0,0) mm`, for example, all 10k muons are thrown from exactly `(0,0,0)`.

Reply:
I assume also the muons were all the same charge? Such that if you flipped the charge the plot of loca vs phi would basically be inverted

Reply:
I only used negative muons in the above simulations.

I'm not sure if for positive muons the `loc-a vs. phi` plot would invert, since the globalToLocal determines `Loc-a` only based on the PCA position and the momentum direction, not the charge. So, for positive muons thrown from the same point `(x,y,z) = (1,0,0) mm` and the same `phi`, both the PCA position and direction would be the same (ignoring any bending of the particle between the creation point and the PCA position).

[https://acts.readthedocs.io/en/latest/core/geometry/surfaces.html#line-surface](https://acts.readthedocs.io/en/latest/core/geometry/surfaces.html#line-surface)

Reply:
If I instead throw 10k positive muons from `(x,y,z) = (1,0,0) mm`, I see the plot below. So the shape is the same for both charge values.

<img width="521" alt="image" src="https://github.com/eic/EICrecon/assets/8701169/56770e05-920e-4c69-9128-ca8bdbae18f1">

There are some seeds with phi reconstructed very poorly due to the charge being incorrectly determined, as was shown in one of the Track Reconstruction meetings. And this 'distortion' is at different phi values for positive and negative muons.

Reply:
What is the global x-y distribution of the seeds? Is this "modulation" distribution a result of the way we project the seeds back to the (0,0) x-y origin and then transform to the local coordinate on the perigee? Or is this modulation a result of the actual PCA calculation in the global to local transform?

Reply:
The modulation comes from the projection of the particle to the point of closest approach w.r.t the line surface, as shown in the diagram below:

<img width="597" alt="pic0" src="https://github.com/eic/EICrecon/assets/8701169/024eb45a-0b5e-46e3-ae7c-01cce1c0c9eb">

The sign of `Loc-a` is then given by whether the particle is going clockwise or counterclockwise around the line surface. This can be seen in the code here:

https://github.com/acts-project/acts/blob/main/Core/src/Surfaces/LineSurface.cpp#L80-L123

So, if the particle is generated at `(x,y,z) = (+1,0,0) mm`, then it will have the modulation shown above, since positive `phi` corresponds to counterclockwise rotation.

If the particle is generated at `(x,y,z) = (-1,0,0) mm`, then positive `phi` means clockwise rotation and a positive 'Loc-a':

<img width="523" alt="pic1" src="https://github.com/eic/EICrecon/assets/8701169/99cd48f7-c841-4abe-9d4f-60df8e1db669">

Reply:
I think that the Acts code linked above explains why we always see `globalToLocal` succeed once we use the correct `direction` vector, and why the results are the same irrespective of the `tolerance` we set. 

When we fit the seed triplet, we determine the circle parameters -- radius and center. Then we use that to determine the coordinates of the PCA. We set the momentum direction as tangential to the circle at the PCA. So, by construction, the `seed` direction vector fulfills the `globalToLocal` tangential requirement.

Reply:
Okay, thanks for all this digging. I understand your point now and I think that essentially solves the issue of what is going on. So to wrap this PR up and get it merged, I think there are two things that need to be done
1. The tolerance can be removed completely, since as you showed by construction we have no dependence on it since the seed momentum vector is by construction tangential to the circle and thus perpendicular to the line surface at the global PCA provided.
2. Change the setting of  the local position by default to some -9999 values to just an uninitialized 2vector. If the global to local transform fails, then we just discard the seed since by construction it should never fail as we already determine the PCA where Acts expects it to be (with the proper momentum vector). So we could do something like 
```
if(!local.ok())
{
  continue;
}

localpos = local.value();
....
````

Does that seem like a reasonable path forward?

Reply:
Yes, sounds good. I'll implement this.

Reply:
I made the updates. I confirmed that the new results are equivalent to those shown in the tables/plots in this thread.

Reply:
@osbornjd Could you review and approve this, at your convenience?